pkg/proto/types.go

```diff
@@ -4033,6 +4033,22 @@ type fieldsHashesJS struct {
 	LeaseBalanceHash  DigestWrapped `json:"leaseBalanceHash"`
 }
 
+func (fh FieldsHashes) EqualWith(other FieldsHashes) bool {
```

Comment:
Looks complicated. Why not just compare all fields one by one? And I think you can compare `Digest`s without using `bytes.Equal` because they are fixed arrays.